src/jamun/model/energy.py

```diff
@@ -218,99 +203,97 @@ def xhat_normalized(self, y: torch_geometric.data.Batch, sigma: float | torch.Te
                 average_squared_distance=self.average_squared_distance,
                 normalization_type=self.normalization_type,
                 sigma_data=self.sigma_data,
-                D=y.pos.shape[-1],
-                device=y.pos.device,
+                D=pos.shape[-1],
+                device=pos.device,
             )
         radial_cutoff = self.effective_radial_cutoff(sigma) / c_in
 
         # Adjust dimensions.
-        c_in = unsqueeze_trailing(c_in, y.pos.ndim - 1)
-        c_skip = unsqueeze_trailing(c_skip, y.pos.ndim - 1)
-        c_out = unsqueeze_trailing(c_out, y.pos.ndim - 1)
+        c_in = unsqueeze_trailing(c_in, pos.ndim - 1)
+        c_skip = unsqueeze_trailing(c_skip, pos.ndim - 1)
+        c_out = unsqueeze_trailing(c_out, pos.ndim - 1)
         c_noise = c_noise.unsqueeze(0)
 
         # Add edges to the graph.
         with torch.cuda.nvtx.range("add_edges"):
-            y = self.add_edges(y, radial_cutoff)
-
-        with torch.cuda.nvtx.range("clone_y"):
-            xhat = y.clone("pos")
+            topology = self.add_edges(pos, topology, radial_cutoff)
 
-        g = functools.partial(self.g, topology=y, c_noise=c_noise, effective_radial_cutoff=radial_cutoff)
+        g = functools.partial(self.g, topology=topology, c_noise=c_noise, effective_radial_cutoff=radial_cutoff)
         h = functools.partial(norm_wrapper, g=g, c_in=c_in, c_skip=c_skip, c_out=c_out)
 
         with torch.cuda.nvtx.range("g"):
-            xhat.pos = torch.compile(xhat_f, disable=not self.use_torch_compile, **self.torch_compile_kwargs)(y.pos, h)
+            model_predictions_fn = torch.compile(
```

Comment:
have you checked that this still works fine with compile?

Reply:
the tests + my training runs seem to be okay!